WordPress/Learn
# Add Search bar to the homepage

Resolves #2744

As the screencast shows, when searching on the homepage, it goes to the newly added template - search-front-page. When switching to another content type, "course" or "lesson", the behavior remains as is - stay in the archive-course, or archive-lesson templates, displaying their own dedicated UI and filters. And it switches back to the search-front-page template when the content type "Any" is chosen.

The unclear and confusing wording of the filter mentioned in #2787 will be addressed in another PR.

Please also feel free to provide feedback on the flow or design. For example like including a label somewhere on the card to allow users to know whether they are looking at a Course or a Lesson, or do we still want the big text on top of the search page when there's already a text functioning similarly under the search bar. @ndiego @kathrynwp @WordPress/meta-design 

| Big text | Text under the search bar |
|-|-|
| ![image](https://github.com/user-attachments/assets/cc13b995-b38c-4818-bc28-dc3b2aa089a5) | ![image](https://github.com/user-attachments/assets/d1f2d55a-2d71-4b07-837f-4da94948cdff) |


## Testing

1. Search on the homepage.
2. Play around with various filters.
3. Search on the Course / Lesson / Learning pathway archive page, and the results should be on the same page (ie, the header, hero section as well as the filters should still be there)

## Screencast

https://github.com/user-attachments/assets/9b88c254-1017-4a1e-859f-289373bb4388
